catalogue/webapp/pages/search/index.tsx

```diff
@@ -31,16 +108,96 @@ export const getServerSideProps: GetServerSideProps<Props | AppErrorProps> =
       return { notFound: true };
     }
 
-    return {
-      props: removeUndefinedProps({
-        serverData,
-        // TODO Harrison to explore what properties we'd want here
-        pageview: {
-          name: 'search',
-          properties: {},
+    const query = context.query;
+    const params = fromQuery(query);
+
+    const defaultProps = removeUndefinedProps({
+      serverData,
+      query,
+
+      // TODO Harrison to explore what properties we'd want here
+      pageview: {
+        name: 'stories',
+        properties: {},
+      },
+    });
+
+    // Stop here if no query has been entered
+    if (!params.query) {
+      return {
+        props: defaultProps,
+      };
+    }
+
+    try {
+      // Helper function as this is repeated across all queries
+      const setData = (categoryName, fetchedData): FetchedDataProps => {
+        // An error shouldn't stop the other results from displaying
+        if (fetchedData.type === 'Error') {
+          console.error(fetchedData.label + ': Error fetching ' + categoryName);
+        }
+
+        return fetchedData && fetchedData.type !== 'Error'
+          ? fetchedData.results
+          : undefined;
+      };
```

Comment:
Having a helper function is good, but maybe not in the body of `getServerSideProps`? Let's move this out to a top-level function.

Reply:
I agree, but I also visually hated everywhere I put it 😄  Think I'll move it to `search/utils`!